examples/copc_simple_loader.html

```diff
@@ -53,91 +53,80 @@
                 );
 
+                const camera = view.camera.camera3D;
+                camera.far = 10.0 * size.length();
                 camera.position.copy(position);
                 camera.lookAt(lookAt);
                 camera.updateProjectionMatrix();
 
                 view.notifyChange(camera);
             }
 
+            function loadCopc(url, options) {
+                if(!url) return;
+                document.getElementById('share').innerHTML = '<a href="' +
+                    location.href.replace(location.search, '') +
+                    '?copc=' + url
+                    + '" target="_blank">Link to share this view</a>';
 
-            function readURL() {
-                const url = document.getElementById('copc_url').value;
-
-                if (url) {
-                    setUrl(url);
-                }
-            }
-
-            function setUrl(url) {
-                if (!url) return;
-
-                const input_url = document.getElementById('copc_url');
-                if (!input_url) return;
-
-                uri.searchParams.set('copc', url);
-                history.replaceState(null, null, `?${uri.searchParams.toString()}`);
-
-                input_url.value = url;
-                load(url);
-            }
-
-
-            function load(url) {
-                const options = {};
-                const urlParams = uri.searchParams
-                urlParams.keys().forEach(key => {
-                    if (key !== 'copc') {
-                        options[key] = parseInt(urlParams.get(key), 10);
-                    }
-                });
                 const source = new itowns.CopcSource({ url });
 
-                if (layer) {
-                    layer.debugUI.destroy()
-                    view.removeLayer('COPC');
-                    view.notifyChange();
-                    layer.delete();
-                }
-
                 const config = {
                     source,
-                    crs: view.referenceCrs,
                     sseThreshold: 2,
                     pointBudget: 3000000,
                     ...options,
                 };
 
-                layer = new itowns.CopcLayer('COPC', config);
-                view.addLayer(layer).then(onLayerReady);
-                layer.whenReady
-                    .then(() => debug.PointCloudDebug.initTools(view, layer, gui));
+                source.whenReady.then(() => {
+                    setView(source.crs);
+                    layer = new itowns.CopcLayer('COPC', config);
+                    view.addLayer(layer).then(onLayerReady);
+                    layer.whenReady
+                        .then(() => debug.PointCloudDebug.initTools(view, layer, gui));
+                })
+            }
+
+            function setUrl(urlCopc) {
+                const url = location.href.replace(location.search, '') +
+                    '?copc=' + urlCopc;
+                location.assign(url);
```

Comment:
URLSearchParams ?

Reply:
It's not a url search param here, but the creation of a new url that will be used for a new window.

Reply:
it's URLSearchParams